### PR TITLE
Add autocomplete component

### PR DESCRIPTION
**Pull request by paulrobertlloyd**
Adds `rigAutocomplete` component. This doesn’t provide an ‘autosuggest’ behaviour, that can be provided in a separate component. This component only allows users to select from a pre-defined list of options.

Options are passed to the `accessible-autocomplete` from params in the Nunjucks component via a set of `data` attributes. For example:

```njk
{{ rigAutocomplete(decorate({
  autoselect: "true",
  displayMenu: "overlay",
  showAllValues: "true",
  label: {
    classes: "govuk-label--l",
    isPageHeading: true,
    text: "Pick a county"
  },
  hint: {
    text: "For example ‘West Sussex’ or ‘Tyne and Wear’"
  },
  items: [
    { text: "Bedfordshire" },
    { text: "Berkshire" },
    { text: "Bristol" },
    …
  ]
}, "county")) }}
```

### CODE REVIEW

File: package/rig/components/autocomplete/_autocomplete.scss [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+  .autocomplete__input--focused {
+    border-color: $govuk-input-border-colour;
+  }
+}
+
+.autocomplete__dropdown-arrow-down {
+  pointer-events: none;
+  z-index: 0;
+}
```
Comment (fofr) [inline package/rig/components/autocomplete/_autocomplete.scss:26]: Can you add a comment to this – why is it needed?

Comment (paulrobertlloyd) [reply package/rig/components/autocomplete/_autocomplete.scss]: Done!

Comment (paulrobertlloyd) [reply package/rig/components/autocomplete/_autocomplete.scss]: …although not sure why need to set `z-index: 0`; without it the arrow disappears.

Comment (fofr) [reply package/rig/components/autocomplete/_autocomplete.scss]: Heh – I totally forgot about this:
https://github.com/DFE-Digital/apply-for-teacher-training-prototype/pull/104

**Pull request merged**